toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -345,3 +350,135 @@ func getAllPackagesFromChroot(imageConnection *ImageConnection) ([]OsPackage, er
+	loaderEntryDir := filepath.Join(chrootDir, "boot", "loader", "entries")
+	if exists, err := file.DirExists(loaderEntryDir); err != nil {
+		return nil, fmt.Errorf("failed while checking if systemd-boot entry dir '%s' exists:\n%w", loaderEntryDir, err)
+	} else if exists {
+		entries, err := extractSystemdBootEntries(loaderEntryDir)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract systemd-boot entries:\n%w", err)
+		}
+		return &CosiBootloader{
+			Type:        BootloaderSystemdBoot,
+			SystemdBoot: &SystemDBoot{Entries: entries},
+		}, nil
+	}
+
+	// Check for standalone UKI .efi files under /boot/efi/EFI/Linux (UKI without config)
+	espDir := filepath.Join(chrootDir, "boot", "efi")
+	cmdlines, err := extractKernelCmdlineFromUkiEfis(espDir, buildDirAbs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract kernel cmdline from UKI .efi files:\n%w", err)
+	}
+
+	if len(cmdlines) > 0 {
+		var entries []SystemDBootEntry
+		for kernelName, cmdline := range cmdlines {
+			efiPath := filepath.Join("/boot/efi/EFI/Linux", fmt.Sprintf("%s.efi", kernelName))
+
+			kernelVersion, err := getKernelVersion(kernelName)
+			if err != nil {
+				return nil, fmt.Errorf("invalid kernel name in UKI file (%s):\n%w", kernelName, err)
+			}
+			entries = append(entries, SystemDBootEntry{
+				Type:    EntryTypeUKIStandalone,
+				Path:    efiPath,
+				Kernel:  kernelVersion,
+				Cmdline: strings.TrimSpace(cmdline),
+			})
+		}
+		return &CosiBootloader{
+			Type:        BootloaderSystemdBoot,
+			SystemdBoot: &SystemDBoot{Entries: entries},
+		}, nil
+	}
+
+	// Fall back to EFI-based GRUB config
+	if GrubConfigSupportExists(imageConnection.Chroot()) {
+		return &CosiBootloader{
+			Type:        BootloaderGrub,
+			SystemdBoot: nil,
+		}, nil
+	}
+
+	return nil, nil
+}
+
+func extractSystemdBootEntries(entryDir string) ([]SystemDBootEntry, error) {
+	files, err := os.ReadDir(entryDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read directory %s:\n%w", entryDir, err)
+	}
+
+	var entries []SystemDBootEntry
+
+	for _, file := range files {
+		if file.IsDir() || !strings.HasSuffix(file.Name(), ".conf") {
+			continue
+		}
+
+		absPath := filepath.Join(entryDir, file.Name())
+		content, err := os.ReadFile(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read %s: %w", absPath, err)
+		}
+
+		entry := SystemDBootEntry{
+			Path: filepath.Join("boot", "loader", "entries", file.Name()),
+		}
+
+		lines := strings.Split(string(content), "\n")
+		for _, line := range lines {
+			line = strings.TrimSpace(line)
+			switch {
+			case strings.HasPrefix(line, "options"):
+				entry.Cmdline = strings.TrimSpace(strings.TrimPrefix(line, "options"))
+			case strings.HasPrefix(line, "linux"):
+				linuxPath := strings.TrimSpace(strings.TrimPrefix(line, "linux"))
+				if kernelVer, err := getKernelVersion(filepath.Base(linuxPath)); err == nil {
+					entry.Kernel = kernelVer
+				}
+			case strings.HasPrefix(line, "uki"):
```

Comment:
The version of systemd-boot in AZL3 doesn't actually support this option. I think it is planned to land in systemd 258